services/construction.go

```diff
@@ -195,10 +200,23 @@ func (s *constructionAPIService) ConstructionDerive(
 		return nil, terr
 	}
 
-	// According to the Rosetta spec:
-	//     Blockchains that require an on-chain action to create an account
-	//     should not implement this method.
-	return nil, ErrNotImplemented
+	var pk signature.PublicKey
+	if err := pk.UnmarshalBinary(request.PublicKey.Bytes); err != nil {
+		loggerCons.Error("ConstructionDerive: malformed public key",
+			"public_key_hex_bytes", hex.EncodeToString(request.PublicKey.Bytes),
+			"err", err,
+		)
+		return nil, ErrMalformedValue
+	}
+
+	resp := &types.ConstructionDeriveResponse{
+		Address: staking.NewAddress(pk).String(),
```

Comment:
This should probably use `MarshalText` instead of `String`. They should return the same thing (the only case being when the address is malformed which can't happen in this case) but it seems more correct to use it explicitly.

Reply:
oh yeah, that'll be better to get an error rather than a placeholder string.